docs(feed): Doc for 2.3 release (#879)
unoplatform/uno.extensions: doc/Overview/Reactive/in-apps.md

---
# Usage in applications of feeds

> The recommended use for feeds is only in models.

Reviewer: Should we define what a model is ?

Author: ```suggestion
The recommended use for feeds is only in _models_.
A _model_ contains the business logic layer required to format data to the view.
```

> The reactive framework allows you to design state-less _models_, focusing on the presentation logic. 

Reviewer: ```suggestion
The Reactive framework allows you to design state-less _models_, focusing on the presentation logic. 
```

They are expected to expose some `IFeed`, `IListFeed`, `IState` or `IListState`.

> A bindable friendly and higly performant _view model_ is then automatically generated by the reactive framework.

Reviewer: ```suggestion
A bindable friendly and higly performant _view model_ is then automatically generated by the Reactive framework.
```

It is this class that will hold the state and which **has to be set as `DataContext` of your page**.

> [!NOTE]

> > The _ViewModel_ of a _model_ will be created when the class name matches the regex "Model$".

Reviewer: ```suggestion
> The _ViewModel_ of a _model_ is created when the class name matches the regex "Model$".
```

> You can customize that behavior using the `[ImplicitBindables("MyApp\.Presentation\.\.*Model$")]` on the assembly
> or the `[ReactiveBindable]` attribute on you model itself.

> [!NOTE]

> > To ease _model_ declaration, public properties will also be accessible on the _view model_ class. 

Reviewer: ```suggestion
> To ease _models_ declaration, public properties are also accessible on the _view model_ class. 
```

> > You can also access to the model itself through the `Model` property.

Reviewer: ```suggestion
> You can also access the model itself through the `Model` property.
```

</uer:FeedView>
```

## Selection

The reactive framework has embedded support of selected item of a `ListFeed` data-bound to a `Selector`, like the `ListView`.

> It means that you don't have to data-bind (and synchronize) the `ListView.SelectedItem`, it will instead be automatically pushed from and to the view.

Reviewer: ```suggestion
It is not required to data-bind (and synchronize) the `ListView.SelectedItem`, it will instead be automatically pushed from and to the view.
```

```xml
<uer:FeedView Source="{Binding Items}">
	<DataTemplate>
		<ListView ItemsSource="{Binding Data}" />
	</DataTemplate>
</uer:FeedView>
```

> And in your model:

Reviewer: ```suggestion
And in the model:
```

```csharp
public IListState<int> Items => ListView
	.Value(this, () => ImmutableList.Create(1, 2, 3, 4, 5))
	.Selection(SelectedItem);

public IState<int> SelectedItem => State.Value(this, () => 3);
```

### Project selected item into another entity

> You can also synchronize the selected item _key_ into another aggregate root object, for instance, assuming `Profile` and `Country` records:

Reviewer: ```suggestion
It is possible to synchronize the selected item _key_ into another aggregate root object, for instance, assuming `Profile` and `Country` records:
```

```csharp
public record Profile(string? FirstName, string? CountryId);

public record Country(string Id, string Name);
```

> You can automatically store the selected item into the `Profile.CountyId` by doing:

Reviewer: ```suggestion
The selected item automatically stored into the `Profile.CountyId` by doing:
```
